File: packages/sdk-go/defenso.go
// Package defenso is a fail-open Defenso middleware for Go net/http-based frameworks.
//
// Alpha scaffold. Talks to https://app.defen.so/api/policy and .../attacks/ingest.
// Never blocks a request when Defenso is unreachable.
package defenso

import (
	"net/http"
	"os"
)

// Config controls Defenso middleware behavior.
type Config struct {
	Token  string
	APIURL string // defaults to https://app.defen.so
}

// Middleware returns a net/http middleware. Alpha: pass-through.
func Middleware(cfg Config) func(http.Handler) http.Handler {
	if cfg.Token == "" {
		cfg.Token = os.Getenv("DEFENSO_TOKEN")
	}
	if cfg.APIURL == "" {
		cfg.APIURL = "https://app.defen.so"
	}
	return func(next http.Handler) http.Handler {
		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
			// TODO: policy check + async attack-log ingest
			next.ServeHTTP(w, r)
		})
	}
}

File: packages/sdk-ruby/lib/defenso.rb
# frozen_string_literal: true

# Defenso — fail-open security middleware for Rack / Rails.
# Alpha scaffold. Full impl: policy fetch + local eval + background attack ingest.
module Defenso
  VERSION = "0.1.0"

  class Middleware
    def initialize(app, token: nil, api_url: "https://app.defen.so")
      @app = app
      @token = token || ENV["DEFENSO_TOKEN"]
      @api_url = api_url
    end

    def call(env)
      # TODO: run policy check; short-circuit on block; async ingest attack log
      @app.call(env)
    end
  end
end
